### PR TITLE
Add more XAxis domain tests

### PR DESCRIPTION
**Pull request by PavelVanecek**
spoiler alert: Recharts ignores this combination of props

### CODE REVIEW

File: test/cartesian/XAxis.spec.tsx [hunk 1]
```diff
@@ -373,190 +373,243 @@ describe('<XAxis />', () => {
...
-          <XAxis dataKey="x" type="number" domain={spy} allowDataOverflow />
-        </BarChart>,
-      );
-      expectXAxisTicks(container, ['-500', '-250', '0', '250', '500']);
-      expect(spy).toHaveBeenCalledTimes(1);
-      expect(spy).toHaveBeenCalledWith([100, 170], true);
+      it('should default to dataMin, dataMax ANYWAY when domain is provided as strings? I do not understand this behaviour.', () => {
+        const { container } = render(
+          <Component>
+            {/* omg recharts just completely ignores this when it's defined as strings */}
```
Comment (ckifer) [inline test/cartesian/XAxis.spec.tsx:529]: I mean, the domain is supposed to be numbers and it isn't. Would you expect it to parse the strings as numbers or maybe throw an error?

File: test/cartesian/XAxis.spec.tsx [hunk 2]
```diff
@@ -590,20 +643,100 @@ describe('<XAxis />', () => {
...
+      it('should display every third tick with interval = 2', () => {
+        const { container } = render(
+          <BarChart width={300} height={300} data={data}>
+            <XAxis dataKey="x" type="number" interval={2} />
+          </BarChart>,
+        );
+        expectXAxisTicks(container, ['0', '135']);
+      });
+
+      it('should add more ticks with tickCount and then reduce them again with interval', () => {
```
Comment (ckifer) [inline test/cartesian/XAxis.spec.tsx:675]: Interval can also be certain string values if we want tests for those I.e "preserveStart", etc.

**Review by ckifer — APPROVED** [review]
Such awesome tests 🚀

File: test/cartesian/XAxis.spec.tsx (hunk 1 above)
Comment (PavelVanecek) [reply test/cartesian/XAxis.spec.tsx:529]: It parses strings elsewhere

File: test/cartesian/XAxis.spec.tsx (hunk 2 above)
Comment (PavelVanecek) [reply test/cartesian/XAxis.spec.tsx:675]: I couldn't find what difference it makes. There's an example in docs where three out of four axes are exactly the same. Can you think of good values that will make a nice test?

Comment (ckifer) [reply test/cartesian/XAxis.spec.tsx:675]: `preserveEnd` always preserves the last tick and then tries to compute ticks that fit from there
`preserveStart` always preserves the first tick and then tries to compute ticks that fit from there
`preserveStartEnd` tries to do both and create ticks that fit in the middle
`equidistantPreserveStart` tries to do the same as `preserveStart` but it doesn't insert random ticks on one side but not the other

https://codesandbox.io/p/sandbox/line-chart-of-different-axis-intervals-d2kjmp?file=%2Fsrc%2FApp.tsx%3A96%2C27

Play with the width of that and you'll see the differences

image

File: test/cartesian/XAxis.spec.tsx (hunk 1 above)
Comment (ckifer) [reply test/cartesian/XAxis.spec.tsx:529]: true that, I think D3 is what parses strings though? At least string data..

Comment (ckifer) [reply test/cartesian/XAxis.spec.tsx:529]: Yep. Although I would always always steer users away from strings if I could either way

**Pull request merged**